components/Python/Scikit-learn/scikit_learn_predictor/main.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import print_function
+
+import argparse
+import sys
+import time
+import os
+import pickle
+import pandas as pd
+import numpy as np
+
+from sklearn.exceptions import NotFittedError
+from parallelm.components import ConnectableComponent
+from parallelm.mlops import mlops as mlops
+from parallelm.mlops.stats.bar_graph import BarGraph
+from parallelm.mlops.stats.multi_line_graph import MultiLineGraph
+from parallelm.mlops.predefined_stats import PredefinedStats
+
+
+class MCenterComponentAdapter(ConnectableComponent):
+    """
+    Adapter for the do_predict
+    """
+    def __init__(self, engine):
+        super(self.__class__, self).__init__(engine)
+
+    def _materialize(self, parent_data_objs, user_data):
+        df_infer_data = parent_data_objs[0]
+        input_model = self._params.get("input-model")
+        return[do_predict(df_infer_data, input_model)]
+
+
+def do_predict(df_infer, input_model):
+    """
+    Perform predictions:
+        a) load scikit-learn model
+        b) on dataset run predict, collect stats assocaited to predictions
+        c) obtain class probability stats
+        d) report stats using mlops APIs
+    """
+    prog_start_time = time.time()
+    mlops.init()
+    model = pickle.load(open(input_model, "rb"))
```

Comment:
Should we "try" model loading. In case model not is ready no to get an exception.